src/huggingface_hub/_commit_api.py

```diff
@@ -137,22 +137,16 @@ class CommitOperationAdd:
     upload_info: UploadInfo = field(init=False, repr=False)
 
     # Internal attributes
-    _upload_mode: Optional[UploadMode] = None  # set to "lfs" or "regular" once known
-    _is_uploaded: bool = False  # set to True once the file has been upload as LFS
-    _is_committed: bool = False  # set to True once the file has been committed
+    _upload_mode: UploadMode = field(init=False, repr=False)  # set to "lfs" or "regular" once known
```

Comment:
Shouldn't it have a default value as well?

Reply:
Based on the `# set to "lfs" or "regular" once known` comment, not defining a default value for it makes more sense IMO.

Also, we are consistent with the `upload_info` attribute this way.

Reply:
I would still prefer to have it set to `None` rather than not being initialized. The difference with `upload_info` is that the upload info is computed in the `__post_init__` method just after initialization, meaning that the value is never undefined for the user. Here we fetch the upload mode later, not at initialization. It's not a big change but it would avoid a `AttributeError: 'CommitOperationAdd' object has no attribute '_upload_mode'` if someone plays with the object (directly accessing the attribute or using any methods like `__eq__`).

So I'd prefer something like this:
```suggestion
    _upload_mode: Optional[UploadMode] = field(init=False, repr=False, default=None)  # set to "lfs" or "regular" once known
```